installer/tue-install-impl.bash

```diff
@@ -1018,11 +1018,34 @@ function tue-install-ros
     mkdir -p "$ROS_PACKAGE_INSTALL_DIR"
 
     local ros_pkg_dir="$ROS_PACKAGE_INSTALL_DIR"/"$ros_pkg_name"
-    local repos_dir="$TUE_REPOS_DIR"/"$src"
-    # replace spaces with underscores
-    repos_dir=${repos_dir// /_}
-    # now, clean out anything that's not alphanumeric or an underscore
-    repos_dir=${repos_dir//[^a-zA-Z0-9\/\.-]/_}
+    local repos_dir
+    if [ "$install_type" = "git" ]
+    then
+        local output
+        output=$(_git_split_url "$src")
+
+        local array
+        read -r -a array <<< "$output"
+        local domain_name=${array[0]}
+        local repo_address=${array[1]}
+        repos_dir="$TUE_REPOS_DIR"/"$domain_name"/"$repo_address"
+        ## temp; Move repo to new location
+        local repos_dir_old="$TUE_REPOS_DIR"/"$src"
+        repos_dir_old=${repos_dir_old// /_}
+        repos_dir_old=${repos_dir_old//[^a-zA-Z0-9\/\.-]/_}
+        if [ -d "$repos_dir_old" ]
+        then
+            tue-install-debug "mv $repos_dir_old $repos_dir"
+            mv "$repos_dir_old" "$repos_dir"
+        fi
+        # temp; end
+    else
+        repos_dir="$TUE_REPOS_DIR"/"$src"
+        # replace spaces with underscores
+        repos_dir=${repos_dir// /_}
+        # now, clean out anything that's not alphanumeric or an underscore
+        repos_dir=${repos_dir//[^a-zA-Z0-9\/\.-]/_}
```

Comment:
For which case will the else block become active? Mercurial?

Reply:
HG, SVN